libraries/AP_HAL_ChibiOS/hwdef/EAH743QF/hwdef.dat

```diff
@@ -0,0 +1,168 @@
+PC13 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+
+
+
+# USB
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# Debug
+PA13 JTMS-SWDIO SWD
+PA14 JTCK-SWCLK SWD
+
+# SPI1 for ICM42688
+PA4 ICM42688_CS CS
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+
+# SPI2 for MAX7456
+PB12 OSD_CS CS
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+
+# SPI3 for W25Q128JVS
+PA15 FLASH_CS CS
+PC10 SPI3_SCK SPI3
+PC11 SPI3_MISO SPI3
+PB2  SPI3_MOSI SPI3
+
+# I2C1 for Baro_BMP390
+PB8 I2C1_SCL I2C1 PULLUP
+PB9 I2C1_SDA I2C1 PULLUP
+
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1) ANALOG(10)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1) ANALOG(11)
+
+# define default battery setup
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 11.1
+define HAL_BATT_CURR_SCALE 100
+define HAL_BATT_MONITOR_DEFAULT 4
+
+# USART1 - GPS
+PB7  USART1_RX USART1
+PB6  USART1_TX USART1
+define DEFAULT_SERIAL1_PROTOCOL SerialProtocol_GPS
+
+
+# USART2 - OSD HD
+PA2  USART2_TX USART2
+PA3  USART2_RX USART2
+
+# USART3 - OSD Analog
+PB11 USART3_RX USART3
+PB10 USART3_TX USART3
+
+# UART4 - SIK (GCS)
+PA0  UART4_TX UART4 NODMA
+PA1  UART4_RX UART4 NODMA
+define DEFAULT_SERIAL4_PROTOCOL SerialProtocol_RCIN
+
+# UART5
+PD2  UART5_RX UART5 NODMA
+PC12 UART5_TX UART5 NODMA
+
+# USART6 - RX-only from HD_VTX
+PC7  USART6_RX USART6 NODMA
+PC6  USART6_TX USART6 NODMA
+
+# Motors
+PA9  TIM1_CH2  TIM1  PWM(1) GPIO(50) BIDIR  # ESC1
+PA8  TIM1_CH1  TIM1  PWM(2) GPIO(51)  # ESC2
+PC9  TIM3_CH4  TIM3  PWM(3) GPIO(52) BIDIR # ESC3
+PC8  TIM3_CH3  TIM3  PWM(4) GPIO(53)  # ESC4
+PE6  TIM15_CH2 TIM15 PWM(5) GPIO(54)  # ESC5
+PE5  TIM15_CH1 TIM15 PWM(6) GPIO(55)  # ESC6
+
+# LEDs
+PC15 LED0 OUTPUT HIGH GPIO(90) # LED_STAT
+define HAL_LED_ON 1
+
+DMA_PRIORITY TIM1* TIM3* SPI1*
+DMA_NOSHARE *
+
+define HAL_STORAGE_SIZE 16384
+# use last 2 pages for flash storage
+# H743 has 16 pages of 128k each
+STORAGE_FLASH_PAGE 14
+
+# enable logging to dataflash
+define HAL_LOGGING_DATAFLASH_DRIVER AP_Logger_W25NXX
+define HAL_LOGGING_DATAFLASH_ENABLED 1
+
+# spi devices
+SPIDEV icm42688  SPI1 DEVID1 ICM42688_CS    MODE3  2*MHZ  16*MHZ
+SPIDEV osd       SPI2 DEVID4 OSD_CS         MODE0 10*MHZ 10*MHZ
+SPIDEV dataflash SPI3 DEVID2 FLASH_CS       MODE3 104*MHZ 104*MHZ
+
+
+
+# no built-in compass and no external I2C so no compass
+define ALLOW_ARM_NO_COMPASS
+define HAL_I2C_INTERNAL_MASK 0
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+
+# one IMU
+IMU Invensensev3 SPI:icm42688 ROTATION_ROLL_180_YAW_90
+define HAL_DEFAULT_INS_FAST_SAMPLE 1
+
+# one BARO BMP390 (uses BMP388 driver)
+BARO BMP388 I2C:0:0x76
+
+# setup for OSD
+define OSD_ENABLED 1
+define HAL_OSD_TYPE_DEFAULT 1
+ROMFS_WILDCARD libraries/AP_OSD/fonts/font*.bin
+
+
+# disable scripting - no SD card
+define AP_SCRIPTING_ENABLED 0
+
+
+
+# unused
+PB1  INT_GYRO INPUT PULLUP GPIO(93)
+PE1  INT_BARO INPUT PULLUP GPIO(94)
+
+
+# TODO
+# ====================================
+PA10 CC INPUT PULLUP GPIO(91)
```

Comment:
need to add RELAY2 and RELAY3 defines...I assume PIT is the 10V supply control and CC just controls the CC pin